src/Microsoft.AspNetCore.Diagnostics.HealthChecks/HealthCheckOptions.cs

```diff
@@ -10,9 +14,23 @@ namespace Microsoft.AspNetCore.Diagnostics.HealthChecks
     /// </summary>
     public class HealthCheckOptions
     {
+        public IDictionary<HealthCheckStatus, int> ResultStatusCodes { get; } = new Dictionary<HealthCheckStatus, int>()
+        {
+            { HealthCheckStatus.Healthy, StatusCodes.Status200OK },
+            { HealthCheckStatus.Degraded, StatusCodes.Status200OK },
+            { HealthCheckStatus.Unhealthy, StatusCodes.Status503ServiceUnavailable },
+
+            // This means that a health check failed, so 500 is appropriate. This is an error.
+            { HealthCheckStatus.Failed, StatusCodes.Status500InternalServerError },
+        };
+
         /// <summary>
-        /// Gets or sets the path at which the Health Check results will be available.
+        /// Gets or sets a delegate used to write the response.
         /// </summary>
-        public PathString Path { get; set; }
+        /// <remarks>
+        /// The default value is a delegate that will write a minimal <c>text/plain</c> response with the value
+        /// of <see cref="CompositeHealthCheckResult.Status"/> as a string.
+        /// </remarks>
+        public Func<HttpContext, CompositeHealthCheckResult, Task> ResponseWriter { get; set; } = HealthCheckResponseWriters.WriteMinimalPlaintext;
```

Comment:
If we expect users to author these, should this be an interface?

Reply:
We're specifically not introducing an abstraction here. When/if someday we have formatters at this level of the stack, this delegate will be used to plug that in.